### PR TITLE
Materialize Buckets

### PR DESCRIPTION
**Pull request by asutula**
Ok spun my wheels on this for a bit, but the solution was pretty simple and clean. Basically, you can still manually set flags on `tt bucket init` to existing thread and key, or now, you can just pass the `--existing` flag and it will interactively prompt you to choose a remote bucket for whatever org you're using (org is set with the `--org` flag like before)

### CODE REVIEW

File: cmd/tt/bucket.go [hunk 1]
```diff
@@ -76,6 +78,62 @@ Existing configs will not be overwritten.
...
+					}
+					bucketInfos = append(bucketInfos, bucketInfo{ID: id, Name: name, Key: root.Key})
+				}
+			}
+
+			prompt := promptui.Select{
+				Label: "Which exiting Bucket do you want to init from?",
+				Items: bucketInfos,
+				Templates: &promptui.SelectTemplates{
+					Active:   fmt.Sprintf(`{{ "%s" | cyan }} {{ .Name | bold }} {{ .Key | faint | bold }}`, promptui.IconSelect),
```
Comment (asutula) [inline cmd/tt/bucket.go:121]: in the case of an unnamed bucket, not sure which is best to display as a backup way to differentiate buckets... key or thread id?

**Comment by asutula** [comment]
We can still make a sub command that simple prints your bucket thread ids and keys so you can use them as flags, but this is a simpler ux for now. Let me know if I should add that, though.

File: cmd/tt/bucket.go (hunk 1 above)
Comment (sanderpick) [reply cmd/tt/bucket.go:121]: I'd say key as it's unique to the bucket

File: cmd/tt/main.go [hunk 2]
```diff
@@ -58,6 +58,10 @@ var (
 			Key:      "thread",
 			DefValue: "",
 		},
+		"existing": {
```
Comment (sanderpick) [inline cmd/tt/main.go:61]: Should this live on in the bucket config?

Comment (asutula) [reply cmd/tt/main.go:61]: it's only used for determining how to create the local bucket, so it doesn't need to be saved anywhere after that. I tried just making it a normal flag not bound with viper, but had some problem. I can try that again.

Comment (sanderpick) [reply cmd/tt/main.go:61]: It should work if you put the flag definition _after_ calling _cmd.BindFlags_

Comment (asutula) [reply cmd/tt/main.go]: All good, I think previously I was getting the flag value from viper instead of `c`. All fixed now.

**Pull request merged**